broker-daemon/models/order.js

```diff
@@ -153,11 +157,67 @@ class Order {
    * @return {Object} Object of parameters the relayer expects
    */
   get paramsForCreate () {
-    const { baseSymbol, counterSymbol, side, baseAmount, counterAmount, makerBaseAddress, makerCounterAddress } = this
+    const {
+      baseSymbol,
+      counterSymbol,
+      side,
+      baseAmount,
+      counterAmount,
+      makerBaseAddress,
+      makerCounterAddress
+    } = this
 
-    return { baseSymbol, counterSymbol, side, baseAmount, counterAmount, makerBaseAddress, makerCounterAddress }
+    return {
+      baseSymbol,
+      counterSymbol,
+      side,
+      baseAmount,
+      counterAmount,
+      makerBaseAddress,
+      makerCounterAddress
+    }
   }
 
+  /**
+   * Params required to place an order on the relayer
+   * @return {Object} Object of parameters the relayer expects
+   */
+  get paramsForPlace () {
+    const {
+      feePaymentRequest,
+      feeRequired,
+      depositPaymentRequest,
+      depositRequired,
+      orderId,
+      outboundSymbol
+    } = this
+
+    if (feeRequired && !feePaymentRequest) {
+      throw new Error(`paramsForPlace: feePaymentRequest is missing.`)
+    }
+
+    if (depositRequired && !depositPaymentRequest) {
+      throw new Error(`paramsForPlace: depositPaymentRequest is missing.`)
+    }
+
+    if (!orderId) {
+      throw new Error(`paramsForPlace: orderId is missing.`)
+    }
+
+    if (!outboundSymbol) {
+      throw new Error(`paramsForPlace: outboundSymbol is missing.`)
+    }
+
+    return {
+      feePaymentRequest,
+      feeRequired,
```

Comment:
passing these fields to the relayer on place or create feels weird. The relayer dictates if the fees/deposits are required and shouldn't have to rely on the broker for this information. Am I misunderstanding this?

Reply:
We don't pass that to the Relayer. I should probably change the docs for this.

We use that flag in the OrderStateMachine to determine whether or not we pay the fee, but we don't send that to the Relayer.